scipy/ndimage/tests/test_morphology.py

```diff
@@ -1149,242 +1313,276 @@ def test_binary_erosion36(self):
                [0, 0, 0, 0, 0, 0, 0, 0],
                [0, 0, 0, 0, 0, 0, 0, 0],
                [0, 0, 0, 0, 0, 0, 0, 1]]
-        data = np.array([[0, 0, 0, 0, 0, 0, 0, 0],
-                         [0, 1, 0, 0, 0, 0, 0, 0],
-                         [0, 0, 0, 0, 0, 1, 1, 1],
-                         [0, 0, 1, 1, 1, 0, 1, 1],
-                         [0, 0, 1, 0, 1, 1, 0, 0],
-                         [0, 1, 0, 1, 1, 1, 1, 0],
-                         [0, 1, 1, 0, 0, 1, 1, 0],
-                         [0, 0, 0, 0, 0, 0, 0, 0]])
-        expected = np.logical_and(tmp, mask)
-        tmp = np.logical_and(data, np.logical_not(mask))
-        expected = np.logical_or(expected, tmp)
+        data = xp.asarray([[0, 0, 0, 0, 0, 0, 0, 0],
+                           [0, 1, 0, 0, 0, 0, 0, 0],
+                           [0, 0, 0, 0, 0, 1, 1, 1],
+                           [0, 0, 1, 1, 1, 0, 1, 1],
+                           [0, 0, 1, 0, 1, 1, 0, 0],
+                           [0, 1, 0, 1, 1, 1, 1, 0],
+                           [0, 1, 1, 0, 0, 1, 1, 0],
+                           [0, 0, 0, 0, 0, 0, 0, 0]])
+        expected = xp.logical_and(tmp, mask)
+        tmp = xp.logical_and(data, xp.logical_not(mask))
+        expected = xp.logical_or(expected, tmp)
         out = ndimage.binary_erosion(data, struct, mask=mask,
                                      border_value=1, origin=(-1, -1))
         assert_array_almost_equal(out, expected)
 
-    def test_binary_erosion37(self):
-        a = np.array([[1, 0, 1],
+    def test_binary_erosion37(self, xp):
+        if is_cupy(xp):
+            pytest.xfail("CuPy: NotImplementedError: only brute_force iteration")
+
+        a = xp.asarray([[1, 0, 1],
                       [0, 1, 0],
                       [1, 0, 1]], dtype=bool)
-        b = np.zeros_like(a)
+        b = xp.zeros_like(a)
         out = ndimage.binary_erosion(a, structure=a, output=b, iterations=0,
                                      border_value=True, brute_force=True)
-        assert_(out is b)
-        assert_array_equal(
+        assert out is b
+        xp_assert_equal(
             ndimage.binary_erosion(a, structure=a, iterations=0,
                                    border_value=True),
             b)
 
-    def test_binary_erosion38(self):
-        data = np.array([[1, 0, 1],
+    def test_binary_erosion38(self, xp):
+        data = xp.asarray([[1, 0, 1],
                         [0, 1, 0],
                         [1, 0, 1]], dtype=bool)
```

Comment:
indentation

Reply:
Since every indentation clean-up is creating a conflict for me in https://github.com/scipy/scipy/pull/21150, I'm going to manually sync the test files from there to here --- when tests actually pass in that one with more than one backend.